internal/driver/controllerserver_helper.go

```diff
@@ -250,11 +253,27 @@ func (cs *ControllerServer) createLinodeVolume(ctx context.Context, label string
 		}
 	}
 
+	// Check encryption status and if it's supported in the specified region.
+	var encryptionStatus string
+	if volumeEncryption == True {
+		supported, err := cs.isEncryptionSupported(ctx, region)
+		if err != nil {
+			return nil, err
+		}
+		if !supported {
+			return nil, errInternal("Volume encryption is not supported in the %s region", region)
+		}
+		encryptionStatus = "enabled"
+	} else {
+		encryptionStatus = "disabled"
+	}
```

Comment:
I think it makes more sense to have this logic inside prepareVolumeParams() helper since we are essentially extracting info from params, validating, and then setting encryption to true.

Reply:
So I tried moving this into prepareVolumeParams() but it's a contradicting change with the previous comment. If we pass req.GetParameters() and handle everything within createAndWaitForVolume(), this parameter will have to be passed into create again. I feel it will make more sense to have this logic in createAndWaitForVolume().

Reply:
Yes it will but it will just be a boolean to say either true or false. We want to move the validation check logic out from here to somewhere else where we are already doing similar checks and extracting data for making the API call. This will keep things consistent and organized